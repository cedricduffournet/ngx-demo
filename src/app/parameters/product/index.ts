export * from './parameter-product.component';
