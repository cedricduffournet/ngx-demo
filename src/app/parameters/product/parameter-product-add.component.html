<app-product-add-view></app-product-add-view>
