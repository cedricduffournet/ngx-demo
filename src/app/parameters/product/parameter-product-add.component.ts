import { ChangeDetectionStrategy, Component, OnInit } from '@angular/core';

@Component({
  selector: 'app-parameter-product-add',
  templateUrl: './parameter-product-add.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class ParameterProductAddComponent implements OnInit {
  constructor() {}

  ngOnInit() {}
}
