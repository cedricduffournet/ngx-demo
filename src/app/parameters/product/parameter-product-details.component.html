<app-product-details-view></app-product-details-view>
