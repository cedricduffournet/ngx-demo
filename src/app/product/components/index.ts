export * from './product-add/product-add.component';
export * from './product-update/product-update.component';
export * from './product-delete/product-delete.component';
export * from './product-form/product-form.component';
export * from './product-item/product-item.component';
export * from './product-items/product-items.component';
export * from './product-details/product-details.component';
