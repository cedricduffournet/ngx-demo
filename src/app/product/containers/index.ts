export * from './product-add-view/product-add-view.component';
export * from './product-update-modal/product-update-modal.component';
export * from './product-delete-modal/product-delete-modal.component';
export * from './product-list-view/product-list-view.component';
export * from './product-selected/product-selected.component';
export * from './product-details-view/product-details-view.component';
