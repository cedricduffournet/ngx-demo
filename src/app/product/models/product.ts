import { schema } from 'normalizr';
import { ProductCategory, productCategorySchema } from '@app/product-category/models/product-category';

export interface Product {
  id: number;
  name: string;
  categories: ProductCategory[];
}

export const productSchema = new schema.Entity('products', {
  categories: [productCategorySchema]
});
