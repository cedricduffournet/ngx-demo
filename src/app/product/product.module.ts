import { NgModule, ComponentFactoryResolver } from '@angular/core';

import { RouterModule } from '@angular/router';
import { EffectsModule } from '@ngrx/effects';
import { StoreModule } from '@ngrx/store';

import { SharedModule } from '@app/shared/shared.module';
import { PipesModule } from '@app/shared/pipes/pipe.module';

import { ProductCategoryModule } from '@app/product-category/product-category.module';
import { ProductEffects } from '@app/product/state/effects';
import { ProductFacade } from '@app/product/state/product.facade';
import {
  ProductAddComponent,
  ProductUpdateComponent,
  ProductDeleteComponent,
  ProductFormComponent,
  ProductItemComponent,
  ProductItemsComponent,
  ProductDetailsComponent
} from '@app/product/components';
import {
  ProductAddViewComponent,
  ProductUpdateModalComponent,
  ProductDeleteModalComponent,
  ProductListViewComponent,
  ProductSelectedComponent,
  ProductDetailsViewComponent
} from '@app/product/containers';
import { reducers } from '@app/product/state/reducers';
import { ProductGuard } from '@app/product/services/product.guard';
import { CoalescingComponentFactoryResolver } from '@app/coalescing-component-factory-resolver.service';

@NgModule({
  imports: [
    RouterModule,
    SharedModule,
    PipesModule,
    ProductCategoryModule,
    EffectsModule.forFeature([ProductEffects]),
    StoreModule.forFeature('products', reducers)
  ],
  declarations: [
    ProductListViewComponent,
    ProductAddComponent,
    ProductUpdateComponent,
    ProductDeleteComponent,
    ProductFormComponent,
    ProductItemComponent,
    ProductItemsComponent,
    ProductAddViewComponent,
    ProductUpdateModalComponent,
    ProductDeleteModalComponent,
    ProductDetailsComponent,
    ProductSelectedComponent,
    ProductDetailsViewComponent
  ],
  entryComponents: [
    ProductUpdateModalComponent,
    ProductDeleteModalComponent
  ],
  exports: [ProductListViewComponent, ProductAddViewComponent, ProductDetailsViewComponent],
  providers: [ProductFacade, ProductGuard]
})
export class ProductModule {
  // see https://github.com/angular/angular/issues/14324
  constructor(
    coalescingResolver: CoalescingComponentFactoryResolver,
    localResolver: ComponentFactoryResolver
  ) {
    coalescingResolver.registerResolver(localResolver);
  }
}
