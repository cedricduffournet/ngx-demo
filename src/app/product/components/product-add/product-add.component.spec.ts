import { TestBed, ComponentFixture } from '@angular/core/testing';
import { ReactiveFormsModule } from '@angular/forms';

import { TranslateModule } from '@ngx-translate/core';
import { FormModule } from '@app/shared/form';
import { ValidationActionModule } from '@app/shared/validation-action';

import {
  ProductAddComponent,
  ProductFormComponent
} from '@app/product/components';
import { Product } from '@app/product/models/product';

describe('ProductAdd', () => {
  let component: ProductAddComponent;
  let fixture: ComponentFixture<ProductAddComponent>;
  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [
        ReactiveFormsModule,
        TranslateModule.forRoot(),
        FormModule,
        ValidationActionModule
      ],
      declarations: [ProductFormComponent, ProductAddComponent]
    });

    fixture = TestBed.createComponent(ProductAddComponent);
    component = fixture.componentInstance;
  });

  it('should be created', () => {
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should emit a save event when form is submitted', () => {
    const product = {
      name: 'Name'
    } as Product;
    spyOn(component.add, 'emit');
    component.onSave(product);

    expect(component.add.emit).toHaveBeenCalledWith(product);
  });

  it('should emit a cancel event when form canceled', () => {
    spyOn(component.cancel, 'emit');
    component.onCancel();

    expect(component.cancel.emit).toHaveBeenCalledWith('cancel');
  });

  it('should indicate to form product is processing', () => {
    component.adding = true;

    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should indicate to form product is not processing', () => {
    component.adding = false;

    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });
});
