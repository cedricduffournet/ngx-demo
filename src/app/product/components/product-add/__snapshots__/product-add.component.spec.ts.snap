// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductAdd should be created 1`] = `
<app-product-add
  add={[Function EventEmitter]}
  adding="false"
  cancel={[Function EventEmitter]}
  categories={[Function Array]}
>
  <app-product-form
    ng-reflect-categories=""
    ng-reflect-processing="false"
  >
    <form
      class="ng-untouched ng-pristine ng-invalid"
      ng-reflect-form="[object Object]"
      novalidate=""
    >
      <section
        class="bg-white"
      >
        <div
          class="p-3"
        >
          <app-text-input
            class="ng-untouched ng-pristine ng-invalid"
            formcontrolname="name"
            ng-reflect-label="NAME"
            ng-reflect-name="name"
            ng-reflect-submitted="false"
          >
            <div
              class="form-group "
            >
              <label
                class="control-label"
              >
                NAME
              </label>
              <input
                class="form-control"
                placeholder=""
                type="text"
              />
              
              
              
              
              
              
            </div>
          </app-text-input>
          
          <app-text-input
            class="ng-untouched ng-pristine ng-valid"
            formcontrolname="amount"
            ng-reflect-label="PRICE"
            ng-reflect-name="amount"
            ng-reflect-submitted="false"
            ng-reflect-type="number"
            type="number"
          >
            <div
              class="form-group "
            >
              <label
                class="control-label"
              >
                PRICE
              </label>
              <input
                class="form-control"
                placeholder=""
                type="number"
              />
              
              
              
              
              
              
            </div>
          </app-text-input>
          <app-text-area
            class="ng-untouched ng-pristine ng-invalid"
            formcontrolname="description"
            ng-reflect-label="DESCRIPTION"
            ng-reflect-name="description"
            ng-reflect-submitted="false"
          >
            <div
              class="form-group "
            >
              <label
                class="control-label"
              >
                DESCRIPTION
              </label>
              <textarea
                class="form-control"
                placeholder=""
                rows="5"
              >
                
    
              </textarea>
              
              
              
            </div>
          </app-text-area>
          
        </div>
      </section>
      <app-wrapper-validation-action>
        <div
          class="d-flex justify-content-end align-items-center p-3"
        >
          <app-validation-button
            ng-reflect-confirm-disabled="false"
            ng-reflect-label-cancel="CANCEL"
            ng-reflect-label-confirm="SAVE"
          >
            <div
              class="button-validation-action text-center d-flex justify-content-end align-items-center p-3-group"
              ng-reflect-klass="button-validation-action text-"
              ng-reflect-ng-class=""
            >
              
              <button
                class="btn btn-cancel btn-lg btn-light"
                ng-reflect-klass="btn btn-cancel btn-lg"
                ng-reflect-ng-class="btn-light"
                type="button"
              >
                 CANCEL 
              </button>
              
              <button
                class="btn btn-confirm btn-lg btn-primary"
                ng-reflect-klass="btn btn-confirm btn-lg"
                ng-reflect-ng-class="btn-primary"
                type="submit"
              >
                
                
                
                <span>
                  SAVE
                </span>
              </button>
            </div>
          </app-validation-button>
        </div>
      </app-wrapper-validation-action>
    </form>
  </app-product-form>
</app-product-add>
`;

exports[`ProductAdd should indicate to form product is not processing 1`] = `
<app-product-add
  add={[Function EventEmitter]}
  adding="false"
  cancel={[Function EventEmitter]}
  categories={[Function Array]}
>
  <app-product-form
    ng-reflect-categories=""
    ng-reflect-processing="false"
  >
    <form
      class="ng-untouched ng-pristine ng-invalid"
      ng-reflect-form="[object Object]"
      novalidate=""
    >
      <section
        class="bg-white"
      >
        <div
          class="p-3"
        >
          <app-text-input
            class="ng-untouched ng-pristine ng-invalid"
            formcontrolname="name"
            ng-reflect-label="NAME"
            ng-reflect-name="name"
            ng-reflect-submitted="false"
          >
            <div
              class="form-group "
            >
              <label
                class="control-label"
              >
                NAME
              </label>
              <input
                class="form-control"
                placeholder=""
                type="text"
              />
              
              
              
              
              
              
            </div>
          </app-text-input>
          
          <app-text-input
            class="ng-untouched ng-pristine ng-valid"
            formcontrolname="amount"
            ng-reflect-label="PRICE"
            ng-reflect-name="amount"
            ng-reflect-submitted="false"
            ng-reflect-type="number"
            type="number"
          >
            <div
              class="form-group "
            >
              <label
                class="control-label"
              >
                PRICE
              </label>
              <input
                class="form-control"
                placeholder=""
                type="number"
              />
              
              
              
              
              
              
            </div>
          </app-text-input>
          <app-text-area
            class="ng-untouched ng-pristine ng-invalid"
            formcontrolname="description"
            ng-reflect-label="DESCRIPTION"
            ng-reflect-name="description"
            ng-reflect-submitted="false"
          >
            <div
              class="form-group "
            >
              <label
                class="control-label"
              >
                DESCRIPTION
              </label>
              <textarea
                class="form-control"
                placeholder=""
                rows="5"
              >
                
    
              </textarea>
              
              
              
            </div>
          </app-text-area>
          
        </div>
      </section>
      <app-wrapper-validation-action>
        <div
          class="d-flex justify-content-end align-items-center p-3"
        >
          <app-validation-button
            ng-reflect-confirm-disabled="false"
            ng-reflect-label-cancel="CANCEL"
            ng-reflect-label-confirm="SAVE"
          >
            <div
              class="button-validation-action text-center d-flex justify-content-end align-items-center p-3-group"
              ng-reflect-klass="button-validation-action text-"
              ng-reflect-ng-class=""
            >
              
              <button
                class="btn btn-cancel btn-lg btn-light"
                ng-reflect-klass="btn btn-cancel btn-lg"
                ng-reflect-ng-class="btn-light"
                type="button"
              >
                 CANCEL 
              </button>
              
              <button
                class="btn btn-confirm btn-lg btn-primary"
                ng-reflect-klass="btn btn-confirm btn-lg"
                ng-reflect-ng-class="btn-primary"
                type="submit"
              >
                
                
                
                <span>
                  SAVE
                </span>
              </button>
            </div>
          </app-validation-button>
        </div>
      </app-wrapper-validation-action>
    </form>
  </app-product-form>
</app-product-add>
`;

exports[`ProductAdd should indicate to form product is processing 1`] = `
<app-product-add
  add={[Function EventEmitter]}
  adding={[Function Boolean]}
  cancel={[Function EventEmitter]}
  categories={[Function Array]}
>
  <app-product-form
    ng-reflect-categories=""
    ng-reflect-processing="true"
  >
    <form
      class="ng-untouched ng-pristine ng-invalid"
      ng-reflect-form="[object Object]"
      novalidate=""
    >
      <section
        class="bg-white"
      >
        <div
          class="p-3"
        >
          <app-text-input
            class="ng-untouched ng-pristine ng-invalid"
            formcontrolname="name"
            ng-reflect-label="NAME"
            ng-reflect-name="name"
            ng-reflect-submitted="false"
          >
            <div
              class="form-group "
            >
              <label
                class="control-label"
              >
                NAME
              </label>
              <input
                class="form-control"
                placeholder=""
                type="text"
              />
              
              
              
              
              
              
            </div>
          </app-text-input>
          
          <app-text-input
            class="ng-untouched ng-pristine ng-valid"
            formcontrolname="amount"
            ng-reflect-label="PRICE"
            ng-reflect-name="amount"
            ng-reflect-submitted="false"
            ng-reflect-type="number"
            type="number"
          >
            <div
              class="form-group "
            >
              <label
                class="control-label"
              >
                PRICE
              </label>
              <input
                class="form-control"
                placeholder=""
                type="number"
              />
              
              
              
              
              
              
            </div>
          </app-text-input>
          <app-text-area
            class="ng-untouched ng-pristine ng-invalid"
            formcontrolname="description"
            ng-reflect-label="DESCRIPTION"
            ng-reflect-name="description"
            ng-reflect-submitted="false"
          >
            <div
              class="form-group "
            >
              <label
                class="control-label"
              >
                DESCRIPTION
              </label>
              <textarea
                class="form-control"
                placeholder=""
                rows="5"
              >
                
    
              </textarea>
              
              
              
            </div>
          </app-text-area>
          
        </div>
      </section>
      <app-wrapper-validation-action>
        <div
          class="d-flex justify-content-end align-items-center p-3"
        >
          <app-validation-button
            ng-reflect-confirm-disabled="true"
            ng-reflect-label-cancel="CANCEL"
            ng-reflect-label-confirm="SAVE"
          >
            <div
              class="button-validation-action text-center d-flex justify-content-end align-items-center p-3-group"
              ng-reflect-klass="button-validation-action text-"
              ng-reflect-ng-class=""
            >
              
              
              <button
                class="btn btn-confirm btn-lg btn-primary"
                disabled=""
                ng-reflect-klass="btn btn-confirm btn-lg"
                ng-reflect-ng-class="btn-primary"
                type="submit"
              >
                
                <span
                  class="spinner-border spinner-border-sm align-baseline"
                  style="font-size: 1rem;"
                />
                
                <span>
                   LOADING
                </span>
                
              </button>
            </div>
          </app-validation-button>
        </div>
      </app-wrapper-validation-action>
    </form>
  </app-product-form>
</app-product-add>
`;
